knora/dsplib/utils/excel_to_json_lists.py

```diff
@@ -277,66 +324,59 @@ def validate_list_with_schema(json_list: str) -> bool:
         True if the list passed validation, False otherwise
     """
     current_dir = os.path.dirname(os.path.realpath(__file__))
-    with open(os.path.join(current_dir, '../schemas/lists-only.json')) as schema:
+    with open(os.path.join(current_dir, "../schemas/lists-only.json")) as schema:
         list_schema = json.load(schema)
 
     try:
         jsonschema.validate(instance=json_list, schema=list_schema)
     except jsonschema.exceptions.ValidationError as err:
         print(err)
         return False
-    print('List passed schema validation.')
+    print("List passed schema validation.")
     return True
 
 
-def prepare_list_creation(
+def _prepare_list_creation(
     excelfolder: str,
     listname: Optional[str],
     comments: dict[str, Any]
 ) -> tuple[dict[str, Any], list[str]]:
     """
-    Creates the list from Excel files that can be used to build a JSON list. Then, creates the root node for the JSON list.
+    This method extracts the names of the Excel files that are in the folder, and creates the root node based on these
+    Excel file names.
 
     Args:
         excelfolder: path to the folder containing the Excel file(s)
         listname: name of the list to be created
         comments: comments for the list to be created
 
     Returns:
-        rootnode: The rootnode of the list as a dictionary
-        excel_files: list of the Excel files to process
+        rootnode: the empty root node of the list, as a dict
+        excel_files: list of the Excel file names to process
     """
-    # reset the global variables before list creation starts
+    # reset the global variables
     global list_of_previous_node_names
     global list_of_lists_of_previous_cell_values
-
     list_of_previous_node_names = []
     list_of_lists_of_previous_cell_values = []
 
-    # check if the given folder parameter is actually a folder
     if not os.path.isdir(excelfolder):
-        print(excelfolder, ' is not a directory.')
-        exit(1)
-
-    # create a list with all excel files from the path provided by the user
-    excel_files = [filename for filename in glob.iglob(f'{excelfolder}/*.xlsx')
-                   if not os.path.basename(filename).startswith('~$')
-                   and os.path.isfile(filename)]
+        raise BaseError(f"ERROR: {excelfolder} is not a directory.")
 
-    # print the files that can be used
-    print('Found the following files:')
-    for file in excel_files:
-        print(file)
```

Comment:
This will be printed in the higher-level methods, because depending on the caller, a different print is needed.